lib/event.c

```diff
@@ -1790,8 +1790,8 @@ struct event *event_fetch(struct event_loop *m, struct event *fetch)
 			tw = thread_timer_wait(&m->timer, &tv);
 
 		if (event_list_count(&m->ready) ||
-		    (tw && !timercmp(tw, &zerotime, >)))
-			tw = &zerotime;
```

Comment:
I'm not sure about this: if there are ready events, I'd prefer to stay with the zero timeout and get going. I do agree it would be fine to set a floor for timer-based timeouts: there's no point in trying for perfect timer granularity.